python/cudf_polars/cudf_polars/callback.py

```diff
@@ -185,9 +185,6 @@ def _callback(
     n_rows: int | None,
     should_time: Literal[False],
     *,
-    device: int | None,
```

Comment:
I've moved all of our configuration usage on to `ConfigOptions`. The user-facing API will still have the split of some things specified at the top-level of `GPUEngine(...)` and other things ending up in `GPUEngine.config`.

Reply:
I had to partially revert this: `ConfigOptions` gets serialized by dask in the streaming executor with the distributed scheduler. Having a `rmm.mr.DeviceMemoryResource` instance on the ConfigOptions made it not serializable.